lib/Trolley/Exception/Standard.php

```diff
@@ -9,28 +9,34 @@
 * @package    Trolley
 * @subpackage Exception
 */
+
 class Standard extends Exception
 {
 	protected $errorBody;
+	private $errorArray;
 
 	/**
 	 * @var $errorBody string|array
 	 */
 	public function __construct($errorBody)
 	{
 		$message = '';
-		if (is_array($errorBody)) {
-			foreach($errorBody as $e) {
-				$message .= "{$e['code']}: {$e['message']}";
-				if (!empty($e['field'])) {
-					$message .= " (field: {$e['field']})";
-				}
-				$message .= "\n";
-			}
-		} elseif (is_string($errorBody)) {
-			$message = $errorBody;
+
+		if (is_string($errorBody)) {
+			$errorBody = json_decode($errorBody, true);
 		}
+
+		$this->errorArray = $errorBody["errors"];
+		$message = json_encode($this->errorArray);
```

Comment:
json_encode the error array might be bad for display. maybe concat the error messages if possible?

Reply:
You mean the string representation that `json_encode()` will print may not have good formatting, so we should continue parsing the array and concatenate by hand?

Reply:
yes - imagine for API caller - it is a bit bizarre assumption that errorMessage would be a json_encoded string as "array(2) {[0] => object { "errorCode" => 400, ...} ... }"

just concat the error message - it would be verbose but more human readable

Reply:
@Yongyiw Okay, I get the idea.
Changed the code to concatenate the error message.

Now, for an error:

`message` values are like:
```PHP
Value is invalid (field: payments.sourceAmount)
Value is invalid (field: payments.sourceAmount)
```
`getAllErrorsAsArray()` will return an array like so:
```PHP
Array
(
    [0] => Array
        (
            [code] => invalid_field
            [message] => Value is invalid
            [field] => payments.sourceAmount
        )

    [1] => Array
        (
            [code] => invalid_field
            [message] => Value is invalid
            [field] => payments.sourceAmount
        )

)
```